src/db/collections/analytics/ledger_size.rs

```diff
@@ -64,30 +59,62 @@ impl OutputCollection {
-                    "total_data_bytes": { "$toString": "$total_data_bytes" },
-                } },
-            ],
-            None,
-        )
-        .await?
-        .try_next()
-        .await?
-        .unwrap_or_default())
+        #[derive(Default, Deserialize)]
+        struct Result {
+            total_storage_deposit_value: d128,
+            total_key_bytes: d128,
+            total_data_bytes: d128,
+            rent_structure: Option<RentStructure>,
+        }
+
+        let res = self
+            .aggregate::<Result>(
+                vec![
+                    doc! { "$match": {
+                        "metadata.booked.milestone_index": { "$lte": ledger_index },
+                        "metadata.spent_metadata.spent.milestone_index": { "$not": { "$lte": ledger_index } }
+                    } },
+                    doc! { "$group" : {
+                        "_id": null,
+                        "total_key_bytes": { "$sum": { "$toDecimal": "$details.rent_structure.num_key_bytes" } },
+                        "total_data_bytes": { "$sum": { "$toDecimal": "$details.rent_structure.num_data_bytes" } },
+                        "total_storage_deposit_value": { "$sum": { "$toDecimal": "$output.storage_deposit_return_unlock_condition.amount" } },
+                    } },
+                    doc! { "$lookup": {
+                        "from": ProtocolUpdateCollection::NAME,
+                        "pipeline": [
+                            { "$match": { "_id": { "$lte": ledger_index } } },
+                            { "$limit": 1 },
+                            { "$replaceWith": "$parameters.rent_structure" }
+                        ],
+                        "as": "rent_structure",
+                    } },
+                    doc! { "$project": {
+                        "total_storage_deposit_value": { "$toString": "$total_storage_deposit_value" },
+                        "total_key_bytes": { "$toString": "$total_key_bytes" },
+                        "total_data_bytes": { "$toString": "$total_data_bytes" },
+                        "rent_structure": { "$first": "$rent_structure" },
+                    } },
+                ],
+                None,
+            )
+            .await?
+            .try_next()
+            .await?
+            .unwrap_or_default();
+
+        Ok(LedgerSizeAnalyticsResult {
+            total_storage_deposit_value: res.total_storage_deposit_value,
+            total_key_bytes: res.total_key_bytes,
+            total_data_bytes: res.total_data_bytes,
+            total_byte_cost: res
+                .rent_structure
+                .map(|rs| {
+                    d128::from(rs.v_byte_cost)
+                        * ((res.total_key_bytes * d128::from(rs.v_byte_factor_key as u32))
+                            + (res.total_data_bytes * d128::from(rs.v_byte_factor_data as u32)))
+                })
+                .unwrap_or_default(),
```

Comment:
I think we should rather `impl iota_types::block::output::Rent for ::Result` and then call `.rent_cost(...)`.